src/include/sof/lib/dma.h

```diff
@@ -116,21 +116,27 @@ struct dma_cb_data {
  * \brief SG elem array.
  */
 struct dma_sg_elem_array {
+	uint32_t id;			/**< e.g. DAI index or stream ID */
+	uint32_t src_dev;		/**< e.g. DMA handshake */
+	uint32_t dest_dev;		/**< e.g. DMA handshake */
 	uint32_t count;			/**< number of elements in elems */
 	struct dma_sg_elem *elems;	/**< elements */
 };
 
+struct dma_sg_array {
+	uint32_t count;				/**< number of elem arrays */
+	struct dma_sg_elem_array *elems;	/**< array of dma_sg elems */
```

Comment:
Should the * elems be a elems[0] ? This way the sg_array is an array that can be allocated on the heap as one item.

Reply:
This might be a good idea, just gotta make sure there aren't any gotchas around the corner.

Reply:
I believe nobody uses sizeof(struct dma_sg_array) and this array isn't sent via IPC either, so it should be fine.

Reply:
I actually don't see any benefit from doing that. Currently, the dma_sg_array exists on the stack and the dma_sg_elem_array is allocated uniformly on the heap. So, it's not going to save us a dereference, since instead of dereferencing elems, we'll end up having to dereference dma_sg_array.